cpp/test/common/sort.cpp

```diff
@@ -26,12 +27,23 @@ TEMPLATE_TEST_CASE("Test radix sort", "[vector][template]", std::int32_t,
   std::generate_n(std::back_inserter(vec), vec_size, generator);
 
   // Sort vector using radix sort
-  dolfinx::radix_sort(std::span(vec));
+  dolfinx::radix_sort(vec);
 
   // Check if vector is sorted
   REQUIRE(std::ranges::is_sorted(vec));
 }
 
+TEST_CASE("Test radix sort (projection)", "[radix]")
+{
+  std::vector<int> vec = {3, 6, 2, 1, 5, 4, 0};
```

Comment:
Switch to a modern int type with specific lengths (and perhaps parametrise over two or three types?).